README.md

```diff
@@ -177,6 +178,38 @@ Listeners available:
 * `on_streampresence` - Receives stream join and leave event.
 * `on_streamdata` - Receives stream data sent by the server.
 
+### Sending match data
+
+Nakama [supports any binary content](https://heroiclabs.com/docs/gameplay-multiplayer-realtime/#send-data-messages) in `data` attribute of a match message. Regardless of your data type, the server **only accepts base64-encoded data**, so make sure you don't post plain-text data or even JSON, or Nakama server will claim the data malformed and disconnect your client (set server logging to `debug` to detect these events).
+
+Here's an example of a proper match data message:
+
+```lua
+local b64 = require "nakama.util.b64"
+local json = require "nakama.util.json"
+
+local data = json.encode({
+    dest_x = 1.0,
+    dest_y = 0.1,
+})
+
+local net_msg = {
+    match_data_send = {
+        match_id = match_id,     -- you get it from on_matchmakermatched() listener in case
+                                 -- of authoritative match, or after joining the relayed match;
+        op_code = 1,             -- pick the op_code for yourself depending on the gameplay event;
+        data = b64.encode(data), -- consider writing a convenience function for encoding your data
```

Comment:
Perhaps this should be part of the nakama-defold extension?

Reply:
Thought it'd be too daring to suggest... :D I'll add it as a util package if you don't mind.